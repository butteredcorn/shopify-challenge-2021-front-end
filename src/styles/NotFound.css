.not-found {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 100vh;
}

.home-link {
    font-size: 1.5rem;
}