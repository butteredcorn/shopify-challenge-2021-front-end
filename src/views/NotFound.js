import React from 'react';
import { Link } from 'react-router-dom';
import '../styles/views/NotFound.css'

const NotFound = () => (
  <div className="not-found">
    <h1>404 - Not Found!</h1>
    <Link to="/" className="home-link">
      Go Home
    </Link>
  </div>
);

export default NotFound;