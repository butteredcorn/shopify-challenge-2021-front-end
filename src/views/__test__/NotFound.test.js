import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import ThemesProvider from '../../providers/ThemeProvider'
import NotFound from '../NotFound';

afterEach(cleanup);

it('renders NotFound component with message "404 - Not Found!"', () => {
  render(<BrowserRouter><ThemesProvider><NotFound/></ThemesProvider></BrowserRouter>);
  const message = screen.getByText(/404 - Not Found!/i);
  expect(message).toBeInTheDocument();
});

it('matches a snapshot', () => {
  const json = renderer.create(<BrowserRouter><ThemesProvider><NotFound/></ThemesProvider></BrowserRouter>).toJSON();
  expect(json).toMatchSnapshot();
});