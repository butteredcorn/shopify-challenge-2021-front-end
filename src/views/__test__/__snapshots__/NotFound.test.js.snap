// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<div
  className="not-found"
>
  <h1>
    404 - Not Found!
  </h1>
  <a
    className="home-link"
    href="/"
    onClick={[Function]}
  >
    Go Home
  </a>
</div>
`;
